src/segment_wifi.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type wifi struct {
+	props          *properties
+	env            environmentInfo
+	Connected      bool
+	State          string
+	SSID           string
+	RadioType      string
+	Authentication string
+	Channel        int
+	ReceiveRate    int
+	TransmitRate   int
+	Signal         int
+}
+
+const (
+	ConnectedIcon    Property = "connected_icon"
+	DisconnectedIcon Property = "disconnected_icon"
+)
+
+const defaultTemplate string = "{{if .Connected}}{{.SSID}} {{.Signal}}% {{.ReceiveRate}}Mbps{{else}}{{.State}}{{end}}"
+
+func (w *wifi) enabled() bool {
+	// If in Linux, who wnows. Gonna need a physical linux machine to develop this since the wlan interface isn't available in wsl
+	// But possible solution is to cat /proc/net/wireless, and if that file doesn't exist then don't show the segment
+	// In Windows and in wsl you can use the command `netsh.exe wlan show interfaces`. Note the .exe is important for this to worw in wsl.
+	if w.env.getPlatform() == windowsPlatform || w.env.isWsl() {
+		cmd := "netsh"
+		if !w.env.hasCommand(cmd) {
+			return false
+		}
+		cmdResult, err := w.env.runCommand(cmd, "wlan", "show", "interfaces")
+		displayError := w.props.getBool(DisplayError, false)
+		if err != nil && displayError {
+			w.State = "WIFI ERR"
+			return true
+		}
+		if err != nil {
+			return false
+		}
+
+		regex := regexp.MustCompile(`(.+) : (.+)`)
```

Comment:
please use the helper functions found in `regex.go`
- [getCompiledRegex](https://github.com/JanDeDobbeleer/oh-my-posh/blob/accf46420ff2924e863f5a76179abc92b2212535/src/regex.go#L15)
- [findNamedRegexMatch](https://github.com/JanDeDobbeleer/oh-my-posh/blob/accf46420ff2924e863f5a76179abc92b2212535/src/regex.go#L35)

Reply:
I wound up just removing the regex altogether and using `strings.Split(" : ")` instead